### DISCUSSION TITLE
Schema is not defined

### DISCUSSION THREAD

**Discussion by ZahidTekbas** [isar/isar#232] [Help] [answered]
**Steps to Reproduce**
I have decided to use [Isar database](https://pub.dev/packages/isar) in my next project and I find it much helpful when dealing with local data.

I followed the [quickstart guide](https://isar.dev/tutorials/quickstart.html) in its website. I added dependencies. Annotated the contact class. Ran code generator. But at fourth step, I have problem creating schema while creating Isar instance.

**Code sample**
```
initIsar() async {
  final dir = await getApplicationSupportDirectory();
  final isar = await Isar.open(
    schemas: [ContactSchema],
    directory: dir.path,
    inspector: true,
  );
}


```
The problem is where I typed ContactSchema, it says
```

Undefined name 'ContactSchema'.
Try correcting the name to one that is defined, or defining the name.
```

So question I wan to ask is, I followed guide but I'm unable to create a schema. How can I create one to make Isar db work?

**Version**
 - Platform: iOS Emulator
 - Flutter 2.8.1 • channel stable
 - Isar version: ^2.1.4
 - Isar Generator version: ^2.1.4

**Comment by simc** [comment]
Did you annotate a model class with `@Collection()`? https://isar.dev/schema.html

**Comment by ZahidTekbas** [comment]
Yes I did it. This is the example class I copied from the guide: 

```
import 'package:isar/isar.dart';

@Collection()
class Contact {
  @Id()
  int? id;

  late String name;
}
```

**Comment by simonpham** [comment, marked as answer]
You need to add `part 'contact.g.dart';` then run `flutter pub build_runner build` again.

```
import 'package:isar/isar.dart';

part 'contact.g.dart';

@Collection()
class Contact {
  @Id()
  int? id;

  late String name;
}
```

**Reply by ZahidTekbas** [reply]
Thank you! That worked.

Additionally, the command is: `flutter pub run build_runner build`

**Comment by simonpham** [comment]
@leisim I think this should be in the documentation.

**Reply by simc** [reply]
It is: https://isar.dev/tutorials/quickstart.html#_3-run-code-generator

Or do you mean the `part 'contact.g.dart';`? The generator output notifies users about that.

**Reply by simonpham** [reply]
I don't know if it's been supported recently. But before when I was using isar version 1.0.5, the generator doesn't notify me about that. It just run success. I had to take a look at the example project to know this.

**Reply by gosemathraj** [reply]
@leisim files are not generated unless you import the generated files. Here "contact.g.dart" is not generated unless we have "part 'contact.g.dart';" in model.

**Reply by simc** [reply]
I know but that has nothing to do with Isar. The build_runner manages that and it is the same for all packages that generate code.

I'll add a small notice to the docs:)
